src/__tests__/to-be-invalid.js

```diff
@@ -0,0 +1,72 @@
+import {JSDOM} from 'jsdom'
+import {render} from './helpers/test-utils'
+
+test('.toBeInvalid', () => {
+  const {queryByTestId} = render(`
+    <div>
+      <input data-testid="no-aria-invalid">
+      <input data-testid="aria-invalid" aria-invalid>
+      <input data-testid="aria-invalid-value" aria-invalid="true">
+      <input data-testid="aria-invalid-false" aria-invalid="false">
+    </div>
+    `)
+
+  const dom = new JSDOM(`<input pattern="[a-z]{1,3}" value="test+">`)
+
+  expect(queryByTestId('no-aria-invalid')).not.toBeInvalid()
+  expect(queryByTestId('aria-invalid')).toBeInvalid()
+  expect(queryByTestId('aria-invalid-value')).toBeInvalid()
+  expect(queryByTestId('aria-invalid-false')).not.toBeInvalid()
+  expect(dom.window.document.querySelector('input')).toBeInvalid()
+
+  // negative test cases wrapped in throwError assertions for coverage.
+  expect(() =>
+    expect(queryByTestId('no-aria-invalid')).toBeInvalid(),
+  ).toThrowError()
+  expect(() =>
+    expect(queryByTestId('aria-invalid')).not.toBeInvalid(),
+  ).toThrowError()
+  expect(() =>
+    expect(queryByTestId('aria-invalid-value')).not.toBeInvalid(),
+  ).toThrowError()
+  expect(() =>
+    expect(queryByTestId('aria-invalid-false')).toBeInvalid(),
+  ).toThrowError()
+  expect(() =>
+    expect(dom.window.document.querySelector('input')).not.toBeInvalid(),
+  ).toThrowError()
+})
+
+test('.toBeValid', () => {
+  const {queryByTestId} = render(`
+    <div>
+      <input data-testid="no-aria-invalid">
+      <input data-testid="aria-invalid" aria-invalid>
+      <input data-testid="aria-invalid-value" aria-invalid="true">
+      <input data-testid="aria-invalid-false" aria-invalid="false">
+    </div>
+    `)
+
+  const dom = new JSDOM(`<input pattern="[a-z]{1,3}" value="test+">`)
```

Comment:
Same comment here. Or probably extract this into a helper function in this very file where you can document the issue just once.

One problem I see with this is that right now tests are runnable in a browser environment. I would be open to merge this and wait to see if this breaks something for someone. At most this would break running the tests for `jest-dom`, not the usage of `jest-dom` itself in other people's tests, right?

I also have an idea based on this. Maybe we can change the implementation of the `render` helper to not rely on `document.innerHTML = ...`, which seems to be part of the problem.